feat(sbb-table-wrapper): initial implementation (#2715)
sbb-design-systems/lyne-components: src/elements/table/table-wrapper/readme.md

> The `sbb-table-wrapper` is a wrapper for a table. Its aim is to enhance/automate tha native table capabilities.

Reviewer: ```suggestion
The `sbb-table-wrapper` is a wrapper for a `<table>`; its goal is to enhance/automate the native table capabilities.
```

> As of now, it only handles overflow (vertical and horizontal).

Reviewer: ```suggestion
Currently it only handles overflow (vertical and horizontal).
```

The `sbb-table-wrapper` is a wrapper for a table. Its goal is to enhance/automate the native table capabilities.

Currently, it only handles overflow (vertical and horizontal).

```html
<sbb-table-wrapper>
  <table class="sbb-table">
    <thead>
      ...
    </thead>
    <tbody>
      ...
    </tbody>
  </table>
</sbb-table-wrapper>
```

## Style

See the [Table](/docs/styles-table--docs) style section.

The component has a `negative` variant which can be set with the self-named property.

```html

> <sbb-table-wrapper negative>

Reviewer 2: Could we pass down the negative property to the table so a consumer doesn't need to set it twice?

Author: I thought we decided not to do that during our call but maybe we should rediscuss it.
Essentially we have three options here:
1. No automation (the consumer sets the negative property and the class)
2. The wrapper checks for the negative class on the table and syncs itself (I don't think we have a consistent way to react to an eventual class change though)
3. The wrapper sets the negative class on the table (The property has priority)

I believe none of them is optimal, maybe the 3rd is best :/
What do you think? @jeripeierSBB @kyubisation

Reviewer 3: Technically we could add the selector `sbb-table-wrapper[negative]` (or maybe `sbb-table-wrapper[negative] .sbb-table`) to the selector `.sbb-table-negative`.

Reviewer 2: But this would still be only top-down and not bottom-up. But yeah it's a simple solution for the top down approach ;)

Author: I like it. Top-down is covered. 
Do we do something for the bottom-up? We could check for the negative class in a @slotchange ( even though it does not cover every case )

Reviewer 3: I wonder if we need the bottom-up case. We could also just document when you are using the sbb-table-wrapper, you need to consider the correct usage.

Reviewer 2: I think documentation is a good starting point :) But as written, I'm almost certain, that some projects will forget to manually set `negative` and nobody will realize it because the outcome is not obviously different. But for sure, we could accept it or start with documentation only and if we see a lot of misusage, we could provide additional logic at a later time.
